code/game/objects/machinery/doors/door.dm

```diff
@@ -245,6 +244,9 @@
 /obj/machinery/door/proc/requiresID()
 	return TRUE
 
+/obj/machinery/door/proc/hasPower()
```

Comment:
just use the bitfield check instead of a proc.

Reply:
/tg/ uses this.